Fix "comparison of Float with nil failed"

rsym opened the pull request:
to fix this ArgumentError

```
2024-07-01 18:29:11 +0900 [warn]: #0 emit transaction failed: error_class=ArgumentError error="comparison of Float with nil failed" location="/var/lib/td-agent/vendor/bundle/ruby/3.1.0/gems/fluent-plugin-numeric-monitor-1.0.3/lib/fluent/plugin/out_numeric_monitor.rb:195:in `>'" tag="*******************"
```

rsym committed:
Fix "comparison of Float with nil failed"

```
2024-07-01 18:29:11 +0900 [warn]: #0 emit transaction failed: error_class=ArgumentError error="comparison of Float with nil failed" location="/var/lib/td-agent/vendor/bundle/ruby/3.1.0/gems/fluent-plugin-numeric-monitor-1.0.3/lib/fluent/plugin/out_numeric_monitor.rb:195:in `>'" tag="*******************"
```

rsym committed:
Merge pull request #1 from rsym/fix_comparison_of_float_with_nil_failed

Fix "comparison of Float with nil failed"

tagomoris commented:
@rsym Could you add the test case for your expected input data?

rsym committed:
Add Test "test_emit_with_nil_values"

rsym commented:
@tagomoris 

I've committed the test case.
Could you please review it again?

```ruby
Started
Finished in 2.605224 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
13 tests, 122 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
4.99 tests/s, 46.83 assertions/s
```

---

When testing with the unmodified `out_numeric_monitor.rb`, we get this result.

```ruby
Started
F
======================================================================================================================================================================================================================
Failure: test_emit_with_nil_values(NumericMonitorOutputTest): comparison of Float with nil failed.
/path/to/fluent-plugin-numeric-monitor/test/plugin/test_out_numeric_monitor.rb:435:in `rescue in test_emit_with_nil_values'
/path/to/fluent-plugin-numeric-monitor/test/plugin/test_out_numeric_monitor.rb:408:in `test_emit_with_nil_values'
     405:     assert_equal 6, r['key_prefix_num']
     406:   end
     407: 
  => 408:   def test_emit_with_nil_values
     409:     d = create_driver(CONFIG)
     410:     begin
     411:       d.run(default_tag: 'test.tag1') do
======================================================================================================================================================================================================================
Finished in 2.545104 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
13 tests, 116 assertions, 1 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
92.3077% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
5.11 tests/s, 45.58 assertions/s
rake aborted!
```

tagomoris commented on test/plugin/test_out_numeric_monitor.rb:
```diff
@@ -404,4 +404,34 @@ def test_output_key_prefix
+      assert_equal 9, r1['tag1_percentile_90']
+      assert_equal 90, r1['tag1_num']
+    end
+  rescue => e
```
Remove this `rescue` clause including `flunk(e)`

tagomoris commented on test/plugin/test_out_numeric_monitor.rb:
```diff
@@ -404,4 +404,34 @@ def test_output_key_prefix
+
+  def test_emit_with_nil_values
+    d = create_driver(CONFIG)
+    begin
```
This `begin` clause doesn't have `rescue` nor `ensure`. Please remove this.

tagomoris commented on test/plugin/test_out_numeric_monitor.rb:
```diff
@@ -404,4 +404,34 @@ def test_output_key_prefix
+    begin
+      d.run(default_tag: 'test.tag1') do
+        10.times do
+          d.feed({'field1' => nil})
```
Does this test need 10 input records?
If just 1 input record is enough to cause the error, use just 1 record.

tagomoris reviewed (changes requested):
The test case has additional/confusing lines.
Could you update the test case?

rsym committed:
remove begin/rescue

rsym committed:
use just 1 record

rsym committed:
Assign nil to feed1 when `c[:min]` and `c[:max]` are not nil.

Fixed `error=“comparison of Float with nil failed”`.
This error occurs when `c[:min]` and `c[:max]` is nil.
To reproduce this situation, nil should be assigned to `feed1` in the second or later position.

tagomoris commented:
Thank you!
I've merged this change and shipped v1.0.4.
